routes/player.js

```diff
@@ -67,22 +68,47 @@ router.get('/song', function(req, res) {
   var searchWords = req.query.song;
   var accessToken = req.session.user.access_token;
   var refreshToken = req.session.user.refresh_token;
+
   if (typeof searchWords === 'string') {
     searchWords = [searchWords];
   }
 
-  spotify.searchSong(searchWords, accessToken)
-  .catch(spotify.OldTokenError, function (err) {
-    // statusCode 401:  Unauthorized
-    return spotify.refreshToken(req.session.user.refresh_token)
-    .then(function (body) {
-      util.saveToken(req, body.access_token, body.refresh_token);
-      return spotify.searchSong(searchWords, body.access_token);
+  var resultSongs = [];
+
+  //first look in database
+  var dbTargetSongs = _.map(searchWords, function(searchWord) {
+    return new RegExp(searchWord, 'i');
+  });
+
+  GhettoNest.find( { $or: [ { title: { $in: dbTargetSongs } }, { artist_name: dbTargetSongs } ] } )
```

Comment:
maybe this should be abstracted somewhere, maybe like a file called 'ghettoNest.js'